radiant_mlhub/models/ml_model.py

```diff
@@ -0,0 +1,120 @@
+"""Extensions of the `PySTAC <https://pystac.readthedocs.io/en/latest/>`_ classes that provide convenience methods for interacting
+with the `Radiant MLHub API <https://docs.mlhub.earth/#radiant-mlhub-api>`_."""
+
+from __future__ import annotations
+
+from datetime import datetime as Datetime
+from typing import Any, Dict, List, Optional, Union, cast
+
+from pystac.catalog import Catalog
+from pystac.collection import Collection
+from pystac.item import Item
+
+from .. import client
+
+
+class MLModel(Item):
+
+    session_kwargs: Dict[str, Any] = {}
+
+    """
+    Class inheriting from :class:`pystac.Item` that adds some convenience methods for listing and fetching from the Radiant MLHub API.
+    """
+    def __init__(  # type: ignore[no-untyped-def]
+        self,
+        id: str,
+        geometry: Optional[Dict[str, Any]],
+        bbox: Optional[List[float]],
+        datetime: Optional[Datetime],
+        properties: Dict[str, Any],
+        stac_extensions: Optional[List[str]] = None,
+        href: Optional[str] = None,
+        collection: Optional[Union[str, Collection]] = None,
+        extra_fields: Optional[Dict[str, Any]] = None,
+        *,
+        api_key=None,
+        profile=None
```

Comment:
```suggestion
    def __init__(
        self,
        id: str,
        geometry: Optional[Dict[str, Any]],
        bbox: Optional[List[float]],
        datetime: Optional[Datetime],
        properties: Dict[str, Any],
        stac_extensions: Optional[List[str]] = None,
        href: Optional[str] = None,
        collection: Optional[Union[str, Collection]] = None,
        extra_fields: Optional[Dict[str, Any]] = None,
        *,
        api_key: Optional[str] = None,
        profile: Optional[str] = None
```

Looks like we don't need to ignore this error if we add type annotations to the `api_key` and `profile` parameters. I'm guessing this code was based on the [`radiant_mlhub.models.Collection.__init__` method](https://github.com/radiantearth/radiant-mlhub/blob/c179d7d65995a30f8d4d9237ab69a13cc994fb8c/radiant_mlhub/models.py#L31), which similarly includes a `# type: ignore` statement. I'm not sure if we needed this for an older version of PySTAC or if I was just being lazy 😊 , but we should probably fix it for the `MLModel` class at least. If you want to fix it in the `Collection` class by adding type annotations feel free, otherwise I can open a separate PR.

Reply:
> If you want to fix it in the Collection class by adding type annotations feel free, otherwise I can open a separate PR.

I'll do that now @duckontheweb